spec-tests/src/test/java/org/candlepin/spec/activationkey/ActivationKeySpecTest.java

```diff
@@ -0,0 +1,651 @@
+import org.candlepin.spec.bootstrap.data.builder.ProductAttributes;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@SpecTest
+public class ActivationKeySpecTest {
+
+    @Test
+    public void shouldAllowActivationKeyFieldFiltering() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        ActivationKeyDTO activationKey = createActivationKey(adminClient, owner);
+
+        ActivationKeyDTO keyFull = adminClient.activationKeys().getActivationKey(activationKey.getId());
+        ActivationKeyDTO keyFiltered = Request.from(adminClient)
+            .setPath("/activation_keys/{activation_key_id}")
+            .setPathParam("activation_key_id", activationKey.getId())
+            .addQueryParam("exclude", "name")
+            .execute()
+            .deserialize(ActivationKeyDTO.class);
+
+        assertThat(keyFull)
+            .usingRecursiveComparison()
+            .ignoringFields("created")
+            .ignoringFields("updated")
+            .ignoringFields("name")
+            .isEqualTo(keyFiltered);
+        assertThat(keyFiltered)
+            .hasFieldOrPropertyWithValue("name", null);
+    }
+
+    @Test
+    public void shouldSetAutoAttachOnActivationKey() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+
+        ActivationKeyDTO activationKey = createActivationKey(
+            adminClient, owner, ActivationKeys.random(owner).autoAttach(null));
+        assertThat(activationKey)
+            .hasFieldOrPropertyWithValue("autoAttach", null);
+
+        ActivationKeyDTO keyWithAutoAttach = createActivationKey(
+            adminClient, owner, ActivationKeys.random(owner).autoAttach(true));
+        assertThat(keyWithAutoAttach)
+            .hasFieldOrPropertyWithValue("autoAttach", true);
+
+        ActivationKeyDTO keyWithoutAutoAttach = createActivationKey(
+            adminClient, owner, ActivationKeys.random(owner).autoAttach(false));
+        assertThat(keyWithoutAutoAttach)
+            .hasFieldOrPropertyWithValue("autoAttach", false);
+    }
+
+    @Test
+    public void shouldAllowOwnersToListExistingActivationKeys() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+
+        createActivationKey(adminClient, owner);
+        createActivationKey(adminClient, owner);
+        createActivationKey(adminClient, owner);
+
+        List<ActivationKeyDTO> listOfActivationKeys =
+            adminClient.owners().ownerActivationKeys(owner.getKey(), null);
+        assertThat(listOfActivationKeys)
+            .hasSize(3);
```

Comment:
You could also assert against the listOfActivationKeys to make sure that they equal the created ActivationKeys.

Reply:
done